libraries/AP_L1_Control/AP_L1_Control_Heli.cpp

```diff
@@ -0,0 +1,549 @@
+#include <AP_HAL/AP_HAL.h>
+#include "AP_L1_Control_Heli.h"
+
+extern const AP_HAL::HAL& hal;
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_L1_Control_Heli::var_info[] = {
```

Comment:
how much of this new file is exactly the same as the existing? Can it be changed to incorporate into the existing? Also, we could have an option to have this as an L1 of enum type two. For Nav controllers, we only have a single controller

Reply:
@magicrub the big problem with using the Navigation.h in combination with the L1 is that I'm stove piped to specific calls.  I will need to add new calls to use new methods I developed for Heli use.  When I spoke with @tridge about this, he had indicated that the navigation and spdhgt headers were developed for plane when it had more than one nav controller.  I would vote to get rid of these header files.  Then map plane directly to the L1 and TECS.  I would then wrap around the L1 with the heli L1 where I would add the heli specific methods/calls.  I don't know at the moment if I can even use anything from the TECS because planes and helis control the x and z axes differently.